### PR TITLE
feat(coap): Support both PKI and PSK being configured concurrently (IEC-78)

### PR DESCRIPTION
**Pull request by mrdeep1**
# Checklist

- [ ] Component contains License
- [ ] Component contains README.md
- [ ] Component contains idf_component.yml file with `url` field defined
- [ ] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [ ] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [ ] _Optional:_ Component contains unit tests
- [ ] CI passing

# Change description

Addresses #285

### CODE REVIEW

File: coap/Kconfig [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
-    choice COAP_MBEDTLS_ENCRYPTION_MODE
-        prompt "CoAP Encryption method"
-        default COAP_MBEDTLS_PSK
+    config COAP_MBEDTLS_PSK
+        bool "Pre-Shared Keys"
+        default y
         help
             If the CoAP information is to be encrypted, the encryption environment
-            can be set up in one of two ways (default being Pre-Shared key mode)
+            can be set up uaing Pre-Shared keys (PSK) mode.
```
Comment (hmalpani) [inline coap/Kconfig:7]:
```suggestion
            can be set up using Pre-Shared keys (PSK) mode.
```

File: coap/Kconfig [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
-            bool "Pre-Shared Keys"
-
-        config COAP_MBEDTLS_PKI
-            bool "PKI Certificates"
+    config COAP_MBEDTLS_PKI
+        bool "PKI Certificates"
+        default n
+        help
+            If the CoAP information is to be encrypted, the encryption environment
+            can be set up uaing Public Key Infrastructure (PKI) mode.
```
Comment (hmalpani) [inline coap/Kconfig:19]:
```suggestion
            can be set up using Public Key Infrastructure (PKI) mode.
```

File: coap/examples/coap_client/main/coap_client_example_main.c [hunk 3]
```diff
@@ -355,13 +384,23 @@ static void coap_example_client(void *p)
         goto clean_up;
 #endif /* CONFIG_MBEDTLS_TLS_CLIENT */
 
+        /*
+         * Encrypted request.  Client needs to chose whether using PSK (if configured)
```
Comment (hmalpani) [inline coap/examples/coap_client/main/coap_client_example_main.c:388]:
```suggestion
         * Encrypted request.  Client needs to choose whether using PSK (if configured)
```

File: coap/Kconfig (hunk 1 above)
Comment (mrdeep1) [reply coap/Kconfig:7]: Corrected.

File: coap/Kconfig (hunk 2 above)
Comment (mrdeep1) [reply coap/Kconfig:19]: Corrected.

File: coap/examples/coap_client/main/coap_client_example_main.c (hunk 3 above)
Comment (mrdeep1) [reply coap/examples/coap_client/main/coap_client_example_main.c]: Corrected. 3 changes pushed.

**Comment by hmalpani** [comment]
Thanks for updating the PR. I think the certificates also needs to updated. I am getting these logs with only PKI enabled:
```
I (12442) CoAP_client:    0.0.0.0:60719 <-> 20.47.97.44:5684 DTLS: The certificate is not correctly signed by the trusted CA: issue 0x8: 'cf-ca' depth 1
I (12452) CoAP_client: *  0.0.0.0:60719 <-> 20.47.97.44:5684 DTLS: netif: sent   15 bytes
I (12462) CoAP_client: do_mbedtls_handshake: session establish returned 'X509 - Certificate verification failed, e.g. CRL, CA or signature check failed'
I (12472) CoAP_client: ***EVENT: COAP_EVENT_DTLS_ERROR
```

**Comment by mrdeep1** [comment]
Will check.  Looks like it is californium.eclipseprojects.io certificates have been updated.  Will try to find the updated ones.

**Comment by mrdeep1** [comment]
Certificates updated.

**Review by hmalpani — APPROVED** [review]
LGTM!

**Comment by hmalpani** [comment]
@mahavirj should we also bump up the version in example [idf_component.yml](https://github.com/espressif/idf-extra-components/blob/master/coap/examples/coap_client/main/idf_component.yml) file?

**Comment by mrdeep1** [comment]
I did bump the version of `idf_component.yml` in this PR.

File: coap/examples/coap_client/main/certs/coap_client.key [hunk 4]
```diff
@@ -1,4 +1,5 @@
 -----BEGIN PRIVATE KEY-----
```
Comment (mahavirj) [inline coap/examples/coap_client/main/certs/coap_client.key:1]: Any specific reason to update the private keys in the examples? I thought updating the server cert should fix the connection problem. @mrdeep1

Comment (mahavirj) [reply coap/examples/coap_client/main/certs/coap_client.key:1]: > A thing to note here is that the client and server are set up to mutually trust each other here. The client checks the server certificates and the server checks the client certificates AND that they are signed by the same CA before proceeding.

Aha, got it. I somehow missed on this point!

**Review by mahavirj — APPROVED** [review]

**Pull request merged**